apps/ledger-live-mobile/src/components/RequiresBLE/hooks/useRequireBluetooth.ts

```diff
@@ -0,0 +1,175 @@
+  const isAndroidEnableLocationHookEnabled =
+    isHookEnabled &&
+    Platform.OS === "android" &&
+    requiredFor === "scanning" &&
+    androidHasBluetoothPermissions === "granted" &&
+    bluetoothServicesState === "enabled" &&
+    androidHasLocationPermission === "granted";
+
+  const {
+    locationServicesState,
+    checkAndRequestAgain: androidEnableLocationCheckAndRequestAgain,
+  } = useAndroidEnableLocation({
+    isHookEnabled: isAndroidEnableLocationHookEnabled,
+  });
+
+  let bluetoothRequirementsState: BluetoothRequirementsState = "unknown";
+  let retryRequestOnIssue = null;
+  let cannotRetryRequest = false;
+
+  let someUnknown = false;
+
+  // Logic to determine the state of the bluetooth requirements, and the action to perform to retry
+  // Because we check if each hook is enabled, we don't need to respect a specific order for the checks
+  if (isAndroidBluetoothPermissionsHookEnabled) {
+    if (androidHasBluetoothPermissions === "denied") {
+      bluetoothRequirementsState = "bluetooth_permissions_ungranted";
+      retryRequestOnIssue =
+        androidBluetoothPermissionsRequestForPermissionsAgain;
+      cannotRetryRequest = androidBluetoothPermissionsNeverAskAgain;
+    } else if (androidHasBluetoothPermissions !== "granted") {
+      someUnknown = true;
+    }
+  }
+
+  if (isAndroidLocationPermissionHookEnabled) {
+    if (androidHasLocationPermission === "denied") {
+      bluetoothRequirementsState = "location_permission_ungranted";
+      retryRequestOnIssue = androidLocationPermissionRequestForPermissionsAgain;
+      cannotRetryRequest = androidLocationPermissionNeverAskAgain;
+    } else if (androidHasLocationPermission !== "granted") {
+      someUnknown = true;
+    }
+  }
+
+  if (isEnableBluetoothHookEnabled) {
+    if (bluetoothServicesState === "disabled") {
+      bluetoothRequirementsState = "bluetooth_disabled";
+      retryRequestOnIssue = enableBluetoothCheckAndRequestAgain;
+    }
+    // Happens on iOS when user did not grant bluetooth permissions
+    // No retry function on iOS because there is currently no way to request bluetooth permissions directly
+    else if (bluetoothServicesState === "unauthorized") {
+      bluetoothRequirementsState = "bluetooth_permissions_ungranted";
+    } else if (bluetoothServicesState !== "enabled") {
+      someUnknown = true;
+    }
+  }
+
+  if (isAndroidEnableLocationHookEnabled && !retryRequestOnIssue) {
+    if (locationServicesState === "disabled") {
+      bluetoothRequirementsState = "location_disabled";
+      retryRequestOnIssue = androidEnableLocationCheckAndRequestAgain;
+    } else if (locationServicesState !== "enabled") {
+      someUnknown = true;
+    }
+  }
+
+  if (
+    isHookEnabled &&
+    !someUnknown &&
+    bluetoothRequirementsState === "unknown"
+  ) {
+    bluetoothRequirementsState = "all_respected";
+  }
+
+  return {
+    bluetoothRequirementsState,
+    retryRequestOnIssue,
+    cannotRetryRequest,
```

Comment:
here maybe we can return `cannotRetryRequest: cannotRetryRequest || !retryRequestOnIssue` to ensure that it's always true in case there's no retryRequest function. That should cover cases like the one mentioned above and ensure consistency for all consumers of this hook.

Reply:
Good point !, adding this check on `retryRequestOnIssue`